### ISSUE TITLE
Djangae codebase does not enforce code standard

### ISSUE THREAD

**Issue by seawolf42** [potatolondon/djangae#982]
**Issue description**

The current codebase is has inconsistencies from the coding style recommended in the code guidelines ("PEP-8 with a loose line length of 100 characters"). Tools like `flake8` can enforce this standard.

**Steps to reproduce**

`flake8 djangae`

**Proposed solution**

I suggest performing the following steps as separate pull requests:

1. setup (single pull request)
    1. create a `.flake8` file for the project with the following disabled:
        * E111, E114, E116, E123, E226 (indentation errors)
        * E121, E124, E125, E126, E127, E128, E129 (continuation line indentation)
        * E201, E202 (whitespace after '[' and whitespace before ']')
        * E203 (whitespace before ':')
        * E222 (multiple spaces after operator)
        * E225 (missing whitespace around operator)
        * E231 (no whitespace after ',')
        * E241 (multiple spaces after ':')
        * E261 (not enough whitespace before inline comment)
        * E262 (inline comment should start with '# ')
        * E265 (block comment should start with '#')
        * E266 (too many '#' leading comment)
        * E271, E272 (multiple spaces after/before keyword)
        * E301, E302, E303, E305, E306, W292, W293, W391 (wrong number of blank lines)
        * E401 (multiple imports on one line)
        * E402 (module-level import not at top of file)
        * E501 (line length)
        * E703 (statement ends with semicolon)
        * E704 (multiple statements on one line)
        * E711, E712, E714 (direct compare to False/None should be use if not cond: or if cond is not False/None)
        * E713 (test should be 'not in')
        * E731 (use def instead of lambda function)
        * F401 (module imported but never used)
        * F402 (import shadowed by loop variable)
        * F403 (from X import *, unable to find undefined names)
        * F405 (variable may be undefined)
        * F811 (redefinition of unused symbol)
        * F821 (undefined name)
        * F841 (variable assigned but never used)
        * W291 (trailing whitespace)
        * W503 (line break before binary operator)
    1. add a flake8 step to the test command so that fixed issues are never re-introduced
1. begin fixes (several pull requests over time)
    1. re-enable one item from the list above and fix all occurrences
    1. fixes may be to correct the issue or to # noqa the line if the implementation must/should be as-is
1. adjust editor settings for supported editors (several pull requests over time)
    1. some editors allow for visual feedback from lint tools, set these up to use flake8 on open files, each as a separate pull request:
        1. Visual Studio Code
        1. others...?
    1. create `.editorconfig` file to guide editors on easy-resolution issues:
        * no whitespace at end of line
        * proper whitespace at end of file
        * etc...

**Traceback or other info**

N/A

* Djangae version: 0.9.10 / master@commit
* Django version: N/A
* App Engine SDK version: N/A
* Using Cloud SDK: no

**Comment by davidwtbuxton** [comment]
Hi,

I recommend waiting for a response from @adamalton and / or @Kazade before starting work.

I think the addtion of an `.editorconfig` is a very good idea for future work, once a style has been blessed. Me personally I would discourage a pull request which only made style changes to existing code that wasn't being updated otherwise.

David B.

**Comment by seawolf42** [comment]
Will do. Looking forward to thoughts. I'll do item 3 regardless, as that's useful whether or not we go for a full-fledged enforcement mechanism.

**Comment by Kazade** [comment]
So, like @davidwtbuxton said, I would prefer not to have PRs that just change style only. However I'm happy to accept a PR that adds a .flake8 which is PEP8 but warns for >100 chars, and I'll definitely accept a PR with an .editorconfig :)

**Comment by adamalton** [comment]
+1

**Comment by seawolf42** [comment]
... and also created #985 for .flake8, PR #986.
